FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/fragment/isolation/incontainer/EndPassPhraseSessionInIsolationTest.kt

```diff
@@ -0,0 +1,179 @@
+      device.pressBack()
+    }
+    val keysStorage = KeysStorageImpl.getInstance(getTargetContext())
+
+    //at startup we don't have any passphrases in RAM
+    onView(withText(R.string.pass_phrase_not_provided))
+      .check(matches(isDisplayed()))
+    assertFalse(keysStorage.hasNonEmptyPassphrase(KeyEntity.PassphraseType.RAM))
+
+    //open notification and check we don't have "End Pass Phrase Session" button
+    device.openNotification()
+    device.wait(Until.hasObject(By.text(activePassPhraseSessionLabel)), timeout)
+
+    assertNull(
+      getActionButtonUiObject2(
+        device,
+        activePassPhraseSessionLabel,
+        endPassPhraseSessionLabel
+      )
+    )
+    assertFalse(keysStorage.hasNonEmptyPassphrase(KeyEntity.PassphraseType.RAM))
+
+    //return back to the app and provide a pass phrase
+    device.pressBack()
+    onView(withId(R.id.eTKeyPassword))
+      .perform(
+        clearText(),
+        typeText(TestConstants.DEFAULT_STRONG_PASSWORD),
+        closeSoftKeyboard()
+      )
+    onView(withId(R.id.btnUpdatePassphrase))
+      .perform(click())
+    onView(withId(R.id.tVPassPhraseVerification))
+      .check(matches(withText(getResString(R.string.stored_pass_phrase_matched))))
+      .check(matches(hasTextColor(R.color.colorPrimaryLight)))
+    onView(withId(R.id.btnUpdatePassphrase))
+      .check(matches(not(isDisplayed())))
+    assertTrue(keysStorage.hasNonEmptyPassphrase(KeyEntity.PassphraseType.RAM))
+
+    //open notification and check we have "End Pass Phrase Session" button
+    device.openNotification()
+    device.wait(Until.hasObject(By.text(activePassPhraseSessionLabel)), timeout)
+
+    val actionButton = getActionButtonUiObject2(
+      device,
+      activePassPhraseSessionLabel,
+      endPassPhraseSessionLabel
+    )
+    assertNotNull(actionButton)
+    assertTrue(keysStorage.hasNonEmptyPassphrase(KeyEntity.PassphraseType.RAM))
+    actionButton?.click()
+
+    //return back to the app and check that passphrase was erased
+    device.pressBack()
+    onView(withText(R.string.pass_phrase_not_provided))
+      .check(matches(isDisplayed()))
+    assertFalse(keysStorage.hasNonEmptyPassphrase(KeyEntity.PassphraseType.RAM))
+  }
+
+  private fun getActionButtonUiObject2(
+    device: UiDevice,
+    activePassPhraseSessionLabel: String,
+    endPassPhraseSessionLabel: String
+  ) = device.findObjects(
+    By.res(notificationResourcesName)
+  ).firstNotNullOfOrNull {
+    val isActivePassPhraseSessionItem = it.findObject(By.text(activePassPhraseSessionLabel)) != null
+    if (isActivePassPhraseSessionItem) {
+      val expandButton = it.findObject(
+        By.pkg("com.android.systemui")
+          .clazz("android.widget.Button")
+          .res("android:id/expand_button")
+      )
+
+      if (expandButton != null) {
+        expandButton.click()
+        it.findObject(By.text(endPassPhraseSessionLabel))
+      } else null
+    } else null
+  }
```

Comment:
IMO, better to like as belows.
For now, not sure about function return type and doesn't seem like best idea to use `= device.findObjects` in function implmentation.
Removed unnecessary `if (expandButton != null) {` check too though
What do you think?

```kt
private fun getActionButtonUiObject2(
  device: UiDevice,
  activePassPhraseSessionLabel: String,
  endPassPhraseSessionLabel: String
): UiObject2? {
  val notificationShadeObjects = device.findObjects(By.res(notificationResourcesName))

  return notificationShadeObjects.firstNotNullOfOrNull { notification ->
    if (notification.hasObject(By.text(activePassPhraseSessionLabel))) {
      val expandButton = notification.findObject(
        By.pkg("com.android.systemui")
          .clazz("android.widget.Button")
          .res("android:id/expand_button")
      )

      expandButton?.click()
      notification.findObject(By.text(endPassPhraseSessionLabel))
    } else null
  }
}
```

Reply:
I've added changes. Please relook.